### PR TITLE
Re-enable CA1305

### LINKED ISSUE #34361: Revisit suppression of CA1305 in .editorconfig
Latest preview.7 SDK introduced new C# language features including interpolated string builders, along with which comes new warning cases for CA1305. That warning has been configured as a suggestion in `.editorconfig` for now to unblock the build and should be revisited to ensure all cases are appropriately adjusted.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/dotnet/aspnetcore/issues/34361

### CODE REVIEW

File: src/Http/Http.Extensions/src/RequestDelegateFactory.cs [hunk 1]
```diff
@@ -1211,17 +1211,19 @@ private static void SetPlaintextContentType(HttpContext httpContext)
         private static string BuildErrorMessageForMultipleBodyParameters(FactoryContext factoryContext)
         {
             var errorMessage = new StringBuilder();
-            errorMessage.Append($"Failure to infer one or more parameters.\n");
-            errorMessage.Append("Below is the list of parameters that we found: \n\n");
-            errorMessage.Append($"{"Parameter",-20}|{"Source",-30} \n");
-            errorMessage.Append("---------------------------------------------------------------------------------\n");
+            errorMessage.AppendLine("Failure to infer one or more parameters.");
```
Comment (pranavkm) [inline src/Http/Http.Extensions/src/RequestDelegateFactory.cs:1214]: @halter73 \ @davidfowl was this intentionally using `\n` versus `AppendLine`?

File: src/Http/Http.Extensions/src/RequestDelegateFactory.cs [hunk 2]
```diff
@@ -1211,17 +1211,19 @@ private static void SetPlaintextContentType(HttpContext httpContext)
...
+            errorMessage.AppendLine("Below is the list of parameters that we found: ");
+            errorMessage.AppendLine();
+            errorMessage.AppendLine(FormattableString.Invariant($"{"Parameter",-20}|{"Source",-30}"));
+            errorMessage.AppendLine("---------------------------------------------------------------------------------\n");
```
Comment (BrennanConroy) [inline src/Http/Http.Extensions/src/RequestDelegateFactory.cs:1218]: remove `\n`

Comment (pranavkm) [reply src/Http/Http.Extensions/src/RequestDelegateFactory.cs:1218]:
```suggestion
            errorMessage.AppendLine("---------------------------------------------------------------------------------");
```

**Review by halter73 — APPROVED** [review]

File: src/Http/Http.Extensions/src/RequestDelegateFactory.cs (hunk 1 above)
Comment (halter73) [reply src/Http/Http.Extensions/src/RequestDelegateFactory.cs:1214]: I don't believe so unless @rafikiassumaniMSFT says otherwise.

**Pull request merged**